src/Ship.js

```diff
@@ -0,0 +1,29 @@
+function Ship(itinerary) {
+  this.itinerary = itinerary;
+  this.currentPort = itinerary.ports[0];
+  this.previousPort = null;
+  this.currentPort.addShip(this);
+
+  this.setSail = function () {
+    const itinerary = this.itinerary;
+    const currentPortIndex = itinerary.ports.indexOf(this.currentPort);
+
+    if (currentPortIndex === itinerary.ports.length - 1) {
+      throw new Error("End of itinerary reached");
+    }
```

Comment:
There is a bit of unnecessary variable creation happening here, this could be simplified:

```suggestion
  this.setSail = function() {
    if (this.itinerary.ports.indexOf(this.currentPort) === this.itinerary.ports.length - 1) {
      throw new Error("End of itinerary reached");
    }
```